### ISSUE TITLE
Functionality to automatically create jira issues for accessibility problems reported in suite.

### ISSUE THREAD

**Issue by Jakub-Izbicki** [wttech/aet#415]
### Context of the issue/feature

Enhancement, that aims at adding functionality to AET that would allow automatically creating jira bugs for accessibility issues found on pages in a given suite.

The work would consist of two main parts:

1. Implement code that realises the functionality of extracting and aggregating accessibility problems data. The scope is all issues in a given suite/raport. The issues would have to be categorised in a way to maintain readibility.
One solution is to categorise issues according to [W3C guidelines](https://www.w3.org/TR/WCAG20/#ensure-compat-rsv), i.e by principle and guidelines, and then by tests' urls.
The idea is that e.g  when we would have 20 issues with code 1.2.4, the functionality would create one jira bug and link all the issues in description, rather than create 20 jira bugs.

2. Implementing OAuth into AET to connect to jira api in order to create accessibility issue jira bugs.

If you have suggestions or objections to the above text, please comment.

**Comment by Skejven** [comment]
As the implementation can start with implementing such feature only for the accessibility tests, the design session for that solution should also cover other types of tests (like screenshot, source, w3c ...).
What needs to be designed is te UX part, how user can interact with AET to request creating jira tickets.
Is new report view required for that or this feature could somehow fit in the existing report?

**Comment by Jakub-Izbicki** [comment]
@Skejven As far as I am informed, one suggestion is for a button to be placed somewhere in the upper area of the page, which would allow to create jira bugs for the suite, although I'm not sure whether it should be necessary to give user possibility to create bugs for a specific type of error only (screenshot, source, accessibility etc..), and how it would actually look like.

**Comment by Skejven** [comment]
There are few options now, you can propose some solution i the comment and we can discuss about it here or you can organize a design session with this improvement stakeholders.
I'm just saying that this feature is much more than just raising a jira bug for accessibility. Even if the implementation will start only with accessibility part implemented, I'd like to see the solution for other types of checks.

**Comment by RdkLwg** [comment]
2 features worth considering:
- what about adding labels to jira automatically like `accessibility` or `aet` (we should think about the place where the user could define them)?
- when the issues are reported to JIRA it would be good to get a list of them displayed somewhere in AET

**Comment by RdkLwg** [comment]
Perhaps it's a good idea to mark somehow issues already raised and those new ones - it can be correlated to the improvement raised at [377](https://github.com/Cognifide/aet/issues/377).
